### DISCUSSION TITLE
使用Visual Studio Code 透過SSH 進行遠端程式開發

### DISCUSSION THREAD

**Discussion by lx3211** [1101-GS4719-C/Python-Community-GS4719-C#486] [General]
SSH伺服器
---
- 安裝SSH（**client跟server都要安裝**）
`sudo apt-get install ssh`
`sudo apt-get install openssh-server`
- cmd連線到ssh server
`ssh username@ip`
`exit`

VS code ssh
---
- VS code 安裝擴充 remote SSH
![](https://i.imgur.com/D2JuLDf.png)
- 點左下角><
![](https://i.imgur.com/bHEuhe6.png)
- 配置ssh server的檔案 : `Open SSH Configuration File`
![](https://i.imgur.com/QXRKNOH.png)
![](https://i.imgur.com/xq0Vys6.png)
- 可以按照以下格式連接遠端的ssh server
    - 複數個ssh server增加在同一份檔案中即可
![QV5Hr2B](https://user-images.githubusercontent.com/96198440/158634762-c645b074-ff26-4cfb-935c-c9dd26163c91.jpg)
- 連線至ssh server
![](https://i.imgur.com/awUihnI.png)
- 然後身份驗證完選擇要開啟的目錄就可以進行遠端程式開發了～
- 連線成功會看到左下角:`><SSH:<IP>`

**Comment by badguy361** [comment]
真的好用~

**Comment by minniewei** [comment]
我也推薦可以用xshell 或 termius去連接遠端linux主機呦~都是採用ssh方式而且介面非常簡潔。

**Comment by gnsJhenJie** [comment]
在開發網頁程式的時候很方便！不過在遠端主機的安全性也要多加注意，如果是透過密碼而不是key登錄的，建議要避免暴露於公開網路中，避免被暴力破解

**Comment by yucheng1028** [comment]
非常詳細，真的很讚

**Comment by xkllkx** [comment]
謝謝樓主分享，非常的實用~
